### ISSUE TITLE
Explore: colors on the graph legend are mismatched, when showing loki logs.

### ISSUE THREAD

**Issue by phmayer** [grafana/grafana#38827]
This is just a minor issue. But as it can be a bit irritating when using Explore to look at logs, i thought i open up an issue to let you guys know about it.

**What happened**:
The colors for the levels are mismatched in Explore for logs from Loki.
Here is a screenshot to explain:
![2021-09-02 15_52_00-Explore - Loki - Grafana_edited](https://user-images.githubusercontent.com/39195932/131858725-308cebbe-1f60-4012-b50b-cae75e52fb9f.png)
As you can see `unknown` has the color green in the legend (red circle) and also in that small tooltip-legend that pops up on mouse-over (red underline).  
But looking at the bars in the graph, `unknown` has the color grey (red arrow).  
  
This becomes especially clear once i select `unknown` in the legend, to only show logs of type `unknown` in the graph:  
![2021-09-02 15_54_31-Explore - Loki - Grafana](https://user-images.githubusercontent.com/39195932/131859826-25009da9-83eb-44ac-b3a3-acb6d3ef92d9.png)
As you can see, it only shows grey bars.  

As far as i can tell, the colors in the graph seem to be always correct.  
The same goes fot the list of log-lines below that graph, the colors are matched correctly:  
![screenshot_2](https://user-images.githubusercontent.com/39195932/131863717-ba40d3a3-40b9-424b-96f1-e07b7611eb13.png)

It is just the graph-legends that seem to be incorrect!  
And i assume the issue is: it always colors the fist label green, the second one yellow and so on. Regardless of what the levels are.  
It can be seen in this screenshot:
![screenshot](https://user-images.githubusercontent.com/39195932/131861380-94885066-aa24-4196-b635-c19c7ad3f0e0.png)
Here `error` is the second item in the graph-legend and it is colored yellow. In comparison, in the first screenshot `error` was orange.  
  

**What you expected to happen**:
The colors to be consistent accross the legend and the graph.  
(`info` = green, `unknown`=grey, `warning`=yellow, and so on...)  
I edited the screenshot above to visualize the expected behaviour:  
![screenshot_edited](https://user-images.githubusercontent.com/39195932/131867507-2b0d63e5-439e-4e70-9c0b-e04e22630885.png)

**How to reproduce it (as minimally and precisely as possible)**:
For me: Open up Explore in Grafana, select Loki as Datasource and select about any Host/Job/whatever and then show the logs.

**Anything else we need to know?**:
Not really, i hope this is understandable and i didnt miss anything.  
(Also i hope this has not been reported befor, i looked, but i couldnt find anything)

**Environment**:
- Grafana version: 8.1.2
- Data source type & version: Loki 2.3.0
- OS Grafana is installed on: Docker (on an Ubuntu Server)
- User OS & Browser: Windows, tested it with Chrome and Firefox
- Grafana plugins: none
- Others: -

**Comment by ever4Kenny** [comment]
+1 for this issue.
Reproducible for all sorts of logs( not loki specified issue).

**Comment by louiseponton** [comment]
Hey! 

Adding a new comment in here as it seems the error is still reproducible in grafana version 9.0.2. 

**What happened:**
In the Explore section, color of the log level legend is mismatch with the colors of the bars.
<img width="1141" alt="image" src="https://user-images.githubusercontent.com/43138003/186423139-5a3fb9f7-a4ee-48ce-ab79-c94db2f0bef0.png">

**How to reproduce it (as minimally and precisely as possible):**
Accessing Explore in Grafana, select Loki as Datasource and select about any Host/Job/whatever and then show the logs.

**Anything else we need to know?:**
Nop

**Environment:** 

Grafana version: 9.0.2
Data source type & version: Loki 2.6.0
OS Grafana is installed on: a Kubernetes Cluster (on an DebianStretch OS)
User OS & Browser: macOS Monterey 12.1 & Chrome
Grafana plugins: none
Others: -

**Comment by phmayer** [comment]
I dont have that issue on 9.1.1, have you considered maybe trying to upgrade?  
(although i have to say, i have not had the issue ever since it was fixed in the PR above)

**Comment by mentlak0** [comment]
Hello,

I am having a very similar issue on version 8.5.3. I recently upgraded from 8.2.2

What happened: 
In the explore section when querying logs using Loki data source I am getting the below: 

![image](https://user-images.githubusercontent.com/93769720/188885115-bb0a897f-010e-4c2e-b84b-7c1866cc6ead.png)

Before the upgrade there used to be various different logs and colours for each log on the legend relating to status codes (this is happening to all services queried). 

How to reproduce it (as minimally and precisely as possible):

Accessing explore in Grafana, select Loki datasource and query service.

Anything else we need to know?:

Tried to fix by restarting promtail daemon set on K8s, then restarting Loki pod to try re-generate logs. No luck with this fix. 

Not wanting to upgrade to version 9.0.0 + due to break release and want to test.

Environment:
Grafana Version: 8.5.3
Loki data source: 2.2
OS Grafana is installed on: a Kubernetes Cluster (on an DebianStretch OS)
User OS & Browser: Ubuntu 22.04.1 & firefox

**Comment by Matthias64** [comment]
> ![image](https://user-images.githubusercontent.com/93769720/188885115-bb0a897f-010e-4c2e-b84b-7c1866cc6ead.png)

Hello, 

I faced the same issue and raised it here (https://github.com/grafana/grafana/issues/54863).
Please let me know if you found how to fix it ! 

Thanks

**Comment by mentlak0** [comment]
Hello,

Thanks for this creating a new issue. It is very strange and also very annoying. The logs show respective colour, but not the logs on histogram. What version of Loki are you using?

**Comment by mentlak0** [comment]
Yes, same version. Loki v2.2.0

**Comment by Matthias64** [comment]
I just tried with Loki v2.6.1 and I have the same issue

**Comment by mentlak0** [comment]
That's not good. You're on a high Grafana version to me too. Hopefully they investigate the issue soon.

**Comment by ryester19** [comment]
I am also having problems with the histogram showing all log entries as gray 'log' items with Loki 2.6.1. This issue should really be repoened

**Comment by Sincerite** [comment]
The problem still exists on the latest Grafana and Loki versions.

![image](https://user-images.githubusercontent.com/29306505/200803213-c1872629-b511-4f3d-a6f9-8b604bbdf7d7.png)

**Comment by mentlak0** [comment]
@Sincerite Morning, take a look at this: https://github.com/grafana/grafana/issues/54863

**Comment by Sincerite** [comment]
![image](https://user-images.githubusercontent.com/29306505/200813151-92b87a83-e335-4dc7-a3f5-92019095b7a2.png)

@mentlak0, Actually I've already tested that solution but it's not working in my case.
I've got log line with dynamic level field (not static label). Grafana detects that but graph not.

Also posting my promtail config:

![image](https://user-images.githubusercontent.com/29306505/200813865-160e623e-b475-4160-a4e1-66b9f1f2abe2.png)

Any ideas?
